File: dal-dao-gen/dal_ftp/63/cs/Dao/AccountBalanceLetterDao.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using Arch.Data;
using Arch.Data.DbEngine;
using com.ctrip.dal.test.test4.Entity.DataModel;
using com.ctrip.dal.test.test4.Interface.IDao;

namespace com.ctrip.dal.test.test4.Dao
{
   /// <summary>
    /// 更多DAL接口功能，请参阅DAL Confluence，地址：
    /// http://conf.ctripcorp.com/display/SysDev/Dal+Fx+API
    /// </summary>
    public partial class AccountBalanceLetterDao : IAccountBalanceLetterDao
    {
        readonly BaseDao baseDao = BaseDaoFactory.CreateBaseDao("AccCorpDB_INSERT_1");
        
        /// <summary>
        ///  插入AccountBalanceLetter
        /// </summary>
        /// <param name="accountBalanceLetter">AccountBalanceLetter实体对象</param>
        /// <returns>状态代码</returns>
        public int InsertAccountBalanceLetter(AccountBalanceLetter accountBalanceLetter)
        {
            try
            {
                StatementParameterCollection parameters = new StatementParameterCollection();
                parameters.Add(new StatementParameter{ Name = "@RecordId", Direction = ParameterDirection.InputOutput, DbType = DbType.Int32, Value = accountBalanceLetter.RecordId});
                parameters.Add(new StatementParameter{ Name = "@AccBalanceId", Direction = ParameterDirection.Input, DbType = DbType.Int32, Value = accountBalanceLetter.AccBalanceId});
                parameters.Add(new StatementParameter{ Name = "@FileName", Direction = ParameterDirection.Input, DbType = DbType.AnsiString, Value = accountBalanceLetter.FileName});
                parameters.Add(new StatementParameter{ Name = "@UploadPerson", Direction = ParameterDirection.Input, DbType = DbType.AnsiString, Value = accountBalanceLetter.UploadPerson});
                parameters.Add(new StatementParameter{ Name = "@DescribeInfo", Direction = ParameterDirection.Input, DbType = DbType.AnsiString, Value = accountBalanceLetter.DescribeInfo});
                parameters.Add(new StatementParameter{ Name = "@CreateTime", Direction = ParameterDirection.Input, DbType = DbType.DateTime, Value = accountBalanceLetter.CreateTime});
                parameters.Add(new StatementParameter{ Name = "@DataChange_LastTime", Direction = ParameterDirection.Input, DbType = DbType.DateTime, Value = accountBalanceLetter.DataChange_LastTime});
                parameters.Add(new StatementParameter{ Name = "@return",  Direction = ParameterDirection.ReturnValue});

                baseDao.ExecSp("spA_AccountBalanceLetter_i", parameters);

               accountBalanceLetter.RecordId = (int)parameters["@RecordId"].Value;
                return (int)parameters["@return"].Value;
            }
            catch (Exception ex)
            {
                throw new DalException("调用AccountBalanceLetterDao时，访问Insert时出错", ex);
            }
        }
        
        /// <summary>
        /// 修改AccountBalanceLetter
        /// </summary>
        /// <param name="accountBalanceLetter">AccountBalanceLetter实体对象</param>
        /// <returns>状态代码</returns>
        public int UpdateAccountBalanceLetter(AccountBalanceLetter accountBalanceLetter)
        {
            try
            {
                StatementParameterCollection parameters = new StatementParameterCollection();
                parameters.Add(new StatementParameter{ Name = "@RecordId", Direction = ParameterDirection.Input, DbType = DbType.Int32, Value = accountBalanceLetter.RecordId});
                parameters.Add(new StatementParameter{ Name = "@AccBalanceId", Direction = ParameterDirection.Input, DbType = DbType.Int32, Value = accountBalanceLetter.AccBalanceId});
                parameters.Add(new StatementParameter{ Name = "@FileName", Direction = ParameterDirection.Input, DbType = DbType.AnsiString, Value = accountBalanceLetter.FileName});
                parameters.Add(new StatementParameter{ Name = "@UploadPerson", Direction = ParameterDirection.Input, DbType = DbType.AnsiString, Value = accountBalanceLetter.UploadPerson});
                parameters.Add(new StatementParameter{ Name = "@DescribeInfo", Direction = ParameterDirection.Input, DbType = DbType.AnsiString, Value = accountBalanceLetter.DescribeInfo});
                parameters.Add(new StatementParameter{ Name = "@CreateTime", Direction = ParameterDirection.Input, DbType = DbType.DateTime, Value = accountBalanceLetter.CreateTime});
                parameters.Add(new StatementParameter{ Name = "@DataChange_LastTime", Direction = ParameterDirection.Input, DbType = DbType.DateTime, Value = accountBalanceLetter.DataChange_LastTime});
                parameters.Add(new StatementParameter{ Name = "@return",  Direction = ParameterDirection.ReturnValue});

                baseDao.ExecSp("spA_AccountBalanceLetter_u", parameters);

                return (int)parameters["@return"].Value;
            }
            catch (Exception ex)
            {
                throw new DalException("调用AccountBalanceLetterDao时，访问Update时出错", ex);
            }
        }
        
        /// <summary>
        /// 删除AccountBalanceLetter
        /// </summary>
        /// <param name="accountBalanceLetter">AccountBalanceLetter实体对象</param>
        /// <returns>状态代码</returns>
        public int DeleteAccountBalanceLetter(AccountBalanceLetter accountBalanceLetter)
        {
            try
            {
                StatementParameterCollection parameters = new StatementParameterCollection();
                parameters.Add(new StatementParameter{ Name = "@RecordId", Direction = ParameterDirection.Input, DbType = DbType.Int32, Value = accountBalanceLetter.RecordId});
                parameters.Add(new StatementParameter{ Name = "@return",  Direction = ParameterDirection.ReturnValue});

                baseDao.ExecSp("spA_AccountBalanceLetter_d", parameters);

                return (int)parameters["@return"].Value;
            }
            catch (Exception ex)
            {
                throw new DalException("调用AccountBalanceLetterDao时，访问Delete时出错", ex);
            }
        }
        
        /// <summary>
        /// 删除AccountBalanceLetter
        /// </summary>
        /// <param name="recordId">@RecordId #></param>
        /// <returns>状态代码</returns>
        public int DeleteAccountBalanceLetter(int recordId)
        {
            try
            {
                StatementParameterCollection parameters = new StatementParameterCollection();
                parameters.Add(new StatementParameter{ Name = "@RecordId", Direction = ParameterDirection.Input, DbType = DbType.Int32, Value = recordId});
                parameters.Add(new StatementParameter{ Name = "@return",  Direction = ParameterDirection.ReturnValue});

                baseDao.ExecSp("spA_AccountBalanceLetter_d", parameters);

                return (int)parameters["@return"].Value;
            }
            catch (Exception ex)
            {
                throw new DalException("调用AccountBalanceLetterDao时，访问DeleteAccountBalanceLetter时出错", ex);
            }
        }
        

        /// <summary>
        /// 根据主键获取AccountBalanceLetter信息
        /// </summary>
        /// <param name="recordId"></param>
        /// <returns>AccountBalanceLetter信息</returns>
        public AccountBalanceLetter FindByPk(int recordId )
        {
            try
            {
                return baseDao.GetByKey<AccountBalanceLetter>(recordId);
            }
            catch (Exception ex)
            {
                throw new DalException("调用AccountBalanceLetterDao时，访问FindByPk时出错", ex);
            }
        }

        /// <summary>
        /// 获取所有AccountBalanceLetter信息
        /// </summary>
        /// <returns>AccountBalanceLetter列表</returns>
        public IList<AccountBalanceLetter> GetAll()
        {
            try
            {
                return baseDao.GetAll<AccountBalanceLetter>();
            }
            catch (Exception ex)
            {
                throw new DalException("调用AccountBalanceLetterDao时，访问GetAll时出错", ex);
            }
        }
        
        /// <summary>
        /// 取得总记录数
        /// </summary>
        /// <returns>记录数</returns>
        public long Count()
        {
            try
            {
                String sql = "SELECT count(1) from AccountBalanceLetter  with (nolock)  ";

                object obj = baseDao.ExecScalar(sql);
                long ret = Convert.ToInt64(obj);
                return ret;
            }
            catch (Exception ex)
            {
                throw new DalException("调用AccountBalanceLetterDao时，访问Count时出错", ex);
            }
        }
        
        /// <summary>
        ///  检索AccountBalanceLetter，带翻页
        /// </summary>
        /// <param name="obj">AccountBalanceLetter实体对象检索条件</param>
        /// <param name="pagesize">每页记录数</param>
        /// <param name="pageNo">页码</param>
        /// <returns>检索结果</returns>
        public IList<AccountBalanceLetter> GetListByPage(AccountBalanceLetter obj, int pagesize, int pageNo)
        {
             try
            {
                var dic = new StatementParameterCollection();
                StringBuilder sbSql = new StringBuilder(200);

                
                 //计算ROWNUM
                int fromRownum = (pageNo - 1) * pagesize + 1;
                int endRownum = pagesize * pageNo;
                 sbSql.Append("WITH CTE AS ("); //WITH CTE 开始
                sbSql.Append("select row_number() over(order by RecordId desc ) as rownum, ");
                sbSql.Append(@"RecordId, AccBalanceId, FileName, UploadPerson, DescribeInfo, CreateTime, DataChangeLastTime from AccountBalanceLetter (nolock) ");

                //包含查询条件
                //StringBuilder whereCondition = new StringBuilder();
                //if (!string.IsNullOrEmpty(obj.Name))
                //{
                //    //人名
                //    whereCondition.Append("Where Name like @Name ");
                //    dic.AddInParameter("@Name", DbType.String, "%" + obj.Name + "%");
                //}
                //sbSql.Append(whereCondition);

                sbSql.Append(")"); //WITH CTE 结束

                // 用 CTE 完成分页
                sbSql.Append(@"select RecordId, AccBalanceId, FileName, UploadPerson, DescribeInfo, CreateTime, DataChangeLastTime from CTE Where rownum between @from and @end");
                dic.AddInParameter("@from", DbType.Int32, fromRownum);
                dic.AddInParameter("@end", DbType.Int32, endRownum);
                IList<AccountBalanceLetter> list = baseDao.SelectList<AccountBalanceLetter>(sbSql.ToString(), dic);

                return list;
            }
            catch (Exception ex)
            {
                throw new DalException("调用AccountBalanceLetterDao时，访问GetListByPage时出错", ex);
            }
        }






        
    }
}

File: dal-dao-gen/dal_ftp/63/java/Dao/AccountBalanceLetterDao.java
package com.ctrip.dal.test.test4;

import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Timestamp;
import java.sql.Types;
import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;

import com.ctrip.platform.dal.dao.DalClient;
import com.ctrip.platform.dal.dao.DalClientFactory;
import com.ctrip.platform.dal.dao.DalHints;
import com.ctrip.platform.dal.dao.DalParser;
import com.ctrip.platform.dal.dao.DalTableDao;
import com.ctrip.platform.dal.dao.KeyHolder;
import com.ctrip.platform.dal.dao.StatementParameters;
import com.ctrip.platform.dal.dao.helper.AbstractDalParser;
import com.ctrip.platform.dal.dao.helper.DalRowMapperExtractor;
import com.ctrip.platform.dal.dao.helper.DalScalarExtractor;

public class AccountBalanceLetterDao {
	private static final String DATA_BASE = "AccCorpDB_INSERT_1";
	private static final String COUNT_SQL_PATTERN = "SELECT count(1) from AccountBalanceLetter";
	private static final String ALL_SQL_PATTERN = "SELECT * FROM AccountBalanceLetter";
	private static final String PAGE_SQL_PATTERN = "WITH CTE AS (select *, row_number() over(order by RecordId desc ) as rownum" 
			+" from AccountBalanceLetter (nolock)) select * from CTE where rownum between %s and %s";

	private static final String INSERT_SP_NAME = "spA_AccountBalanceLetter_i";
	private static final String DELETE_SP_NAME = "spA_AccountBalanceLetter_d";
	private static final String UPDATE_SP_NAME = "spA_AccountBalanceLetter_u";
	private static final String RET_CODE = "retcode";
	private static final String UPDATE_COUNT = "update_count";
	
	private DalParser<AccountBalanceLetter> parser = new AccountBalanceLetterParser();
	private DalScalarExtractor extractor = new DalScalarExtractor();
	private DalRowMapperExtractor<AccountBalanceLetter> rowextractor = null;
	private DalTableDao<AccountBalanceLetter> client;
	private DalClient baseClient;

	public AccountBalanceLetterDao() {
		this.client = new DalTableDao<AccountBalanceLetter>(parser);
		this.rowextractor = new DalRowMapperExtractor<AccountBalanceLetter>(parser); 
		this.baseClient = DalClientFactory.getClient(DATA_BASE);
	}

	public AccountBalanceLetter queryByPk(Number id)
			throws SQLException {
		DalHints hints = new DalHints();
		return client.queryByPk(id, hints);
	}

	public AccountBalanceLetter queryByPk(AccountBalanceLetter pk)
			throws SQLException {
		DalHints hints = new DalHints();
		return client.queryByPk(pk, hints);
	}
	
	public long count()  throws SQLException
	{
		StatementParameters parameters = new StatementParameters();
		DalHints hints = new DalHints();
		long result = (Long)this.baseClient.query(COUNT_SQL_PATTERN, parameters, hints, extractor);
		return result;
	}
	
	public List<AccountBalanceLetter> queryByPage(AccountBalanceLetter pk, int pageSize, int pageNo)  throws SQLException {
		if(pageNo < 1 || pageSize < 1) 
			throw new SQLException("Illigal pagesize or pageNo, pls check");
		
        StatementParameters parameters = new StatementParameters();
		DalHints hints = new DalHints();
		
		String sql = "";
		int fromRownum = (pageNo - 1) * pageSize + 1;
        int endRownum = pageSize * pageNo;
		sql = String.format(PAGE_SQL_PATTERN, fromRownum, endRownum);
		return this.baseClient.query(sql, parameters, hints, rowextractor);
	}
	
	public List<AccountBalanceLetter> getAll() throws SQLException
	{
		StatementParameters parameters = new StatementParameters();
		DalHints hints = new DalHints();
		List<AccountBalanceLetter> result = null;
		result = this.baseClient.query(ALL_SQL_PATTERN, parameters, hints, rowextractor);
		return result;
	}

	public int insert(AccountBalanceLetter daoPojo) throws SQLException {
		StatementParameters parameters = new StatementParameters();
		DalHints hints = new DalHints();
		
		String callSql = prepareSpCall(INSERT_SP_NAME, parameters, parser.getFields(daoPojo));

		parameters.registerInOut("@RecordId", Types.INTEGER, daoPojo.getRecordId());

		Map<String, ?> results = baseClient.call(callSql, parameters, hints);

		Integer recordId = (Integer)parameters.get("@RecordId", ParameterDirection.InputOutput).getValue();
		
		return (Integer)results.get(RET_CODE);
	}

	public int delete(AccountBalanceLetter daoPojo) throws SQLException {
		StatementParameters parameters = new StatementParameters();
		DalHints hints = new DalHints();
		
		String callSql = prepareSpCall(DELETE_SP_NAME, parameters, parser.getPrimaryKeys(daoPojo));


		Map<String, ?> results = baseClient.call(callSql, parameters, hints);
		
		return (Integer)results.get(RET_CODE);
	}

	public int update(AccountBalanceLetter daoPojo) throws SQLException {
		StatementParameters parameters = new StatementParameters();
		DalHints hints = new DalHints();
		
		String callSql = prepareSpCall(UPDATE_SP_NAME, parameters, parser.getFields(daoPojo));


		Map<String, ?> results = baseClient.call(callSql, parameters, hints);
		
		return (Integer)results.get(RET_CODE);
	}


	private String prepareSpCall(String SpName, StatementParameters parameters, Map<String, ?> fields) {
		client.addParametersByName(parameters, fields);
		String callSql = client.buildCallSql(SpName, fields.size());
		parameters.setResultsParameter(RET_CODE, extractor);
		parameters.setResultsParameter(UPDATE_COUNT);
		return callSql;
	}
	
	private static class AccountBalanceLetterParser extends AbstractDalParser<AccountBalanceLetter> {
		public static final String DATABASE_NAME = "AccCorpDB_INSERT_1";
		public static final String TABLE_NAME = "AccountBalanceLetter";
		private static final String[] COLUMNS = new String[]{
			"RecordId",
			"AccBalanceId",
			"FileName",
			"UploadPerson",
			"DescribeInfo",
			"CreateTime",
			"DataChange_LastTime",
		};
		
		private static final String[] PRIMARY_KEYS = new String[]{
			"RecordId",
		};
		
		private static final int[] COLUMN_TYPES = new int[]{
			Types.INTEGER,
			Types.INTEGER,
			Types.VARCHAR,
			Types.VARCHAR,
			Types.VARCHAR,
			Types.TIMESTAMP,
			Types.TIMESTAMP,
		};
		
		public AccountBalanceLetterParser() {
			super(DATABASE_NAME, TABLE_NAME, COLUMNS, PRIMARY_KEYS, COLUMN_TYPES);
		}
		
		@Override
		public AccountBalanceLetter map(ResultSet rs, int rowNum) throws SQLException {
			AccountBalanceLetter pojo = new AccountBalanceLetter();
			
			pojo.setRecordId((Integer)rs.getObject("RecordId"));
			pojo.setAccBalanceId((Integer)rs.getObject("AccBalanceId"));
			pojo.setFileName((String)rs.getObject("FileName"));
			pojo.setUploadPerson((String)rs.getObject("UploadPerson"));
			pojo.setDescribeInfo((String)rs.getObject("DescribeInfo"));
			pojo.setCreateTime((Timestamp)rs.getObject("CreateTime"));
			pojo.setDatachangeLasttime((Timestamp)rs.getObject("DataChange_LastTime"));
	
			return pojo;
		}
	
		@Override
		public boolean isAutoIncrement() {
			return true;
		}
	
		@Override
		public Number getIdentityValue(AccountBalanceLetter pojo) {
			return pojo.getRecordId();
		}
	
		@Override
		public Map<String, ?> getPrimaryKeys(AccountBalanceLetter pojo) {
			Map<String, Object> primaryKeys = new LinkedHashMap<String, Object>();
			
			primaryKeys.put("RecordId", pojo.getRecordId());
	
			return primaryKeys;
		}
	
		@Override
		public Map<String, ?> getFields(AccountBalanceLetter pojo) {
			Map<String, Object> map = new LinkedHashMap<String, Object>();
			
			map.put("RecordId", pojo.getRecordId());
			map.put("AccBalanceId", pojo.getAccBalanceId());
			map.put("FileName", pojo.getFileName());
			map.put("UploadPerson", pojo.getUploadPerson());
			map.put("DescribeInfo", pojo.getDescribeInfo());
			map.put("CreateTime", pojo.getCreateTime());
			map.put("DataChange_LastTime", pojo.getDatachangeLasttime());
	
			return map;
		}
	}
}